app/main.py

```diff
@@ -1,2 +1,85 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    load_factor = 2 / 3
+
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(
+            self,
+            key: Hashable,
+            value: Any
+    ) -> None:
+        key_hash = hash(key)
+        items = [key, key_hash, value]
+        if self.length >= round(self.capacity * self.load_factor):
+            self.__hash_table_resize()
+
+        self.__put_key_value_to_a_hash_table(items)
+
+    def __hash_table_resize(self) -> None:
+        template_hash_table = self.hash_table
+        self.capacity *= 2
+        self.hash_table = [None] * self.capacity
+        for items in template_hash_table:
+            if items:
+                self.__put_key_value_to_a_hash_table(
+                    items,
+                    ignore_counter_of_length=True
+                )
+
+    def __put_key_value_to_a_hash_table(
+        self,
+        items: list[Hashable | int | Any],
+        ignore_counter_of_length: bool = False
+    ) -> Any:
+
+        key, key_hash, value = items
+        index_in_hash_table = key_hash % self.capacity
+
+        while True:
```

Comment:
you can put lookup for index in a separate function and reuse it in all your get/set/delete methods